src/fowt_ml/linear_models.py

```diff
@@ -0,0 +1,100 @@
+"""Module to handle linear models."""
+
+from dataclasses import dataclass
+import sklearn.linear_model as lm
+import sklearn.metrics as sm
+from numpy.typing import ArrayLike
+
+
+@dataclass
+class Estimator:
+    name: str
+    func: callable
+    reference: str = "sklearn"
+
+
+@dataclass
+class Metric:
+    name: str
+    func: callable
+    reference: str = "sklearn"
+
+
+class LinearModels:
+    """Class to handle linear models and metrics for comparison."""
+
+    ESTIMATOR_NAMES = {
+        "LinearRegression": lm.LinearRegression(),
+        "RidgeRegression": lm.Ridge(),
+        "LassoRegression": lm.Lasso(),
+        "ElasticNetRegression": lm.ElasticNet(),
+        "LeastAngleRegression": lm.Lars(),
+    }
+    METRICS_NAMES = {
+        "root_mean_squared_error": sm.root_mean_squared_error,
+        "mean_squared_error": sm.mean_squared_error,
+        "r2_score": sm.r2_score,
+        "mean_absolute_error": sm.mean_absolute_error,
+    }
+
+    def __init__(self, estimator: str | Estimator, kwargs: dict = None) -> None:
```

Comment:
Maybe instead of `kwargs` either use `**kwargs` (i.e. with unpacking) or use a more descriptive name (maybe `params`?)?

Also, I am not 100% sure, but if a parameter has `None` as default, this should be made explicit in the type hint:
```suggestion
    def __init__(self, estimator: str | Estimator, kwargs: dict | None = None) -> None:
```

Reply:
it is fixed now.